File: 프로그래머스/lv1/1845. 폰켓몬/폰켓몬.py
# 아이디어 : 최대한 많은 종류의 폰켓몬
# 자료구조 : set 자료형으로 중복 제거
# min으로 비교

def solution(nums):
    return min(len(nums)//2, len(set(nums)))

File: 프로그래머스/1/1845. 폰켓몬/폰켓몬.java
import java.util.*;

class Solution {
    public int solution(int[] nums) {
        
        HashSet<Integer> hashSet = new HashSet<>();
        
        for(int num: nums) {
            hashSet.add(num);
        }
        
        int minValue = Math.min(nums.length / 2, hashSet.size());
        
        return minValue;
    }
}